compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -190,7 +203,53 @@ TEST_F(tServiceComponent, testDelayedComponent_LifeCycle) //DS_TOI_52
   auto ctxt = framework.GetBundleContext();
   auto sRef = ctxt.GetServiceReference<test::Interface2>();
   EXPECT_FALSE(static_cast<bool>(sRef)) << "Service must not be available before it's dependency";
-    
+  
+  #if defined(_WIN32)
+
+    HMODULE hMods[1024];
+    DWORD cbNeeded;
+    unsigned int i;
+
+    HANDLE hProcess = GetCurrentProcess();
+    auto res = EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded);
+    EXPECT_EQ(res, 1)<<"Enumeration failed";
+
+    char szModName[MAX_PATH];
+    std::string result;
+
+    for (i = 0; i < (cbNeeded / sizeof(HMODULE)); i++) 
+    {
+      GetModuleFileNameEx(hProcess, hMods[i], szModName, sizeof(szModName) / sizeof(TCHAR));
```

Comment:
I think GetModuleFileNameEx is meant to be used to get module information for a different process than the current process. GetModuleFileName is more reliable when getting information from the current process. In either case these functions can fail and will return an error. Should probably be checked.

Reply:
Replaced GetModuleFileNameEx with GetModuleFileNameA